### PR TITLE
Separate integration tests and unit tests

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
1. Refactor dependencies integration tests.
2. Remove DB dependencies on unit tests target.
3. Unit test reduce from 13 minutes to 2 minutes 30 seconds

**Why?**
Unit tests should not have dependencies other than testing and mocking libs.

**How did you test it?**
Existing tests.

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: tests/integration/cassandra_test.go [hunk 1]
```diff
@@ -1,160 +0,0 @@
-// The MIT License
```
Comment (yux0) [inline tests/integration/cassandra_test.go:1]: @wxing1292 My understanding is those tests are covered in `./common/persistence/tests`. So I can remove the tests from here?

Comment (wxing1292) [reply tests/integration/cassandra_test.go:1]: if there are no tests referencing these setup, feel free to delete

**Review by wxing1292 — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -59,10 +59,14 @@ endef
 
 TEST_TIMEOUT := 20m
 
+# TODO: INTEG_TEST should be functional tests
 INTEG_TEST_ROOT        := ./host
 INTEG_TEST_XDC_ROOT    := ./host/xdc
 INTEG_TEST_NDC_ROOT    := ./host/ndc
 
+INTEGRATION_TEST_ROOT := ./common/persistence/tests
```
Comment (MichaelSnowden) [inline Makefile:67]: Nit: DB_INTEGRATION_TEST_ROOT

File: common/persistence/tests/cassandra_test.go [hunk 3]
```diff
@@ -135,3 +136,40 @@ func TestCassandraTaskQueueTaskSuite(t *testing.T) {
...
 }
+
+func TestCassandraVisibilityPersistence(t *testing.T) {
+	s := &VisibilityPersistenceSuite{
+		TestBase: persistencetests.NewTestBaseWithCassandra(&persistencetests.TestBaseOptions{}),
+	}
+	suite.Run(t, s)
+}
+
+// TODO: Merge persistence-tests into the tests directory.
```
Comment (MichaelSnowden) [inline common/persistence/tests/cassandra_test.go:147]: Do we need this anymore?

**Review by MichaelSnowden — APPROVED** [review]

File: common/persistence/tests/cassandra_test.go (hunk 3 above)
Comment (yux0) [reply common/persistence/tests/cassandra_test.go:147]: Yes. I did not delete the tests under persistence-tests dir.

File: Makefile (hunk 2 above)
Comment (yux0) [reply Makefile:67]: It also include ES tests and any future dependency integration tests. So I want to just use integration tests.

File: Makefile [hunk 4]
```diff
@@ -59,10 +59,14 @@ endef
 
 TEST_TIMEOUT := 20m
 
+# TODO: INTEG_TEST should be functional tests
 INTEG_TEST_ROOT        := ./host
```
Comment (mindaugasrukas) [inline Makefile:63]: How about renaming this `./host` dir, too, to something like `./host_tests`? It could have a clearer directory name.

**Comment by mindaugasrukas** [comment]
Please ensure those test utils don't end up in the final binary. We should probably add build tags to exclude them from the binary.

File: Makefile (hunk 4 above)
Comment (yux0) [reply Makefile:63]: Will do in the following PR

**Comment by yux0** [comment]
> Please ensure those test utils don't end up in the final binary. We should probably add build tags to exclude them from the binary.

Will migrate to use build tag in the following PR and make sure we don't build the test utils.

File: common/persistence/tests/cassandra_test.go (hunk 3 above)
Comment (alexshtin) [reply common/persistence/tests/cassandra_test.go:147]: May be it is time address this TODO, after you merge this PR.

**Review by alexshtin — APPROVED** [review]

**Pull request merged**